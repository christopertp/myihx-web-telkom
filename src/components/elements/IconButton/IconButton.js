import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './styles.scoped.css';

export default function IconButton(props) {
  const { className, name, onClick, title, to, type } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');
  const icon = <img alt={name} src={`/assets/ic_${name}.svg`} />;

  return to ?
    (<Link className={customClass} title={title} to={to}>{icon}</Link>):
    (<button className={customClass} onClick={onClick} type={type}>{icon}</button>);
}

IconButton.defaultProps = {
  className: '',
  name: '',
  onClick: () => {},
  title: '',
  to: '',
  type: 'button',
};

IconButton.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  onClick: PropTypes.func,
  title: PropTypes.string,
  to: PropTypes.string,
  type: PropTypes.string,
};
