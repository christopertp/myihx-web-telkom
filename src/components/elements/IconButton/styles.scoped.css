.root {
  background: none;
  border: none;
  cursor: pointer;
  display: inline-block;
  height: 1.5rem;
}

.root > img {
  display: block;
  height: 100%;
}
