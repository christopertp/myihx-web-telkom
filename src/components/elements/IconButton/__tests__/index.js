import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import IconButton from '../IconButton';

describe('src/components/elements/IconButton', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<IconButton />);
    expect(tree).toMatchSnapshot();
    IconButton.defaultProps.onClick();
  });

  test('render link', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<IconButton to="/" />);
    expect(tree).toMatchSnapshot();
  });
});
