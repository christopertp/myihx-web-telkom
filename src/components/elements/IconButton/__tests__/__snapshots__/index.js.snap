// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/IconButton render 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  <img
    alt=""
    src="/assets/ic_.svg"
  />
</button>
`;

exports[`src/components/elements/IconButton render link 1`] = `
<Link
  className=""
  title=""
  to="/"
>
  <img
    alt=""
    src="/assets/ic_.svg"
  />
</Link>
`;
